### PR TITLE
Fix issue where field/union types resolve to external types unnecessarily

### PR DESCRIPTION
**Pull request by ctumolosus**
@benwaffle is not possible to compute the FQN for fields because it's not possible to guess its type without looking it up in the service anyways, so went with this approach.

### CODE REVIEW

File: src/utilities/apibuilder/type/service.ts [hunk 1]
```diff
@@ -144,16 +136,16 @@ export class ApiBuilderService {
...
-
-  public findEnumByName(name) {
-    return findTypeByName(this.enums, name);
-  }
-
-  public findUnionByName(name) {
-    return findTypeByName(this.unions, name);
+  public findTypeByName(typeName: string) {
+    // By definition, a field or union type whose name is not fully qualified
+    // implies the type is defined internally, that is such type is not imported.
```
Comment (chrhicks) [inline src/utilities/apibuilder/type/service.ts]: While true, there is always the chance that that changes as the order is probably not enforced anywhere ;)

Comment (ctumolosus) [reply src/utilities/apibuilder/type/service.ts]: the order of what is not enforced?

Comment (chrhicks) [reply src/utilities/apibuilder/type/service.ts]: That internal types preced external types in the list. Just musing about the assumption- not trying to block the change or anything :)

**Comment by benwaffle** [comment]
Was the problem that we were choosing external types over internal when doing this?

```js
service.findModelByName(ast.name) ||
service.findUnionByName(ast.name) ||
service.findEnumByName(ast.name)
```

**Comment by ctumolosus** [comment]
@benwaffle Yep. 

The order there was different:
`InternalModel` > `ExternalModel` > `InternalUnion` > `ExternalUnion` > `InternalEnum` > `ExternalEnum`

**Pull request merged**